### PR TITLE
[ET-999] Add is_ticket_cache_enabled() method that allows turning off ticket object cache and fix in stock cache storage

### PR DESCRIPTION
**Pull request by sc0ttkclark**
[ET-999]

Using a string instead of a true/false value may provide a solution to those encountering cache conflict issues where their tickets/RSVPs may appear as out of stock at seemingly random times while they have plenty of capacity left.

The is_ticket_cache_enabled() method provides a new filterable check on whether to make use of caching for the ticket objects. This is currently used on preventing early returns from cached values. This does not stop the cache sets but they are non-persistent right now anyways.

[ET-999]: https://theeventscalendar.atlassian.net/browse/ET-999

### CODE REVIEW

File: src/Tribe/Ticket_Object.php [hunk 1]
```diff
@@ -1149,6 +1151,23 @@ public function has_meta_enabled() {
...
+		 * Determine whether the ticket has ticket cache enabled.
+		 *
+		 * @since TBD
+		 *
+		 * @return bool Whether the ticket has ticket cache enabled.
+		 */
+		public function is_ticket_cache_enabled() {
+			/**
+			 * Allow filtering whether the ticket has ticket cache enabled.
+			 *
```
Comment (juanfra) [inline src/Tribe/Ticket_Object.php:1165]: Let's add a `@since` to the filter as well.

File: src/Tribe/Ticket_Object.php [hunk 2]
```diff
@@ -1149,6 +1151,23 @@ public function has_meta_enabled() {
...
+			 * @param bool $has_meta  Whether the ticket has ticket cache enabled.
```
Comment (juanfra) [inline src/Tribe/Ticket_Object.php:1166]: Let's fix the var name of this param.

**Review by juanfra — COMMENTED** [review]
Looking good! Found a couple minor things on the docblocks.

File: src/Tribe/Ticket_Object.php [hunk 3]
```diff
@@ -520,16 +520,18 @@ public function is_in_stock() {
 			$cache = tribe( 'cache' );
 			$key   = __METHOD__ . '-' . $this->ID;
 
-			if ( isset( $cache[ $key ] ) ) {
-				return $cache[ $key ];
+			if ( $this->is_ticket_cache_enabled() && isset( $cache[ $key ] ) ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Ticket_Object.php:523]:
```suggestion
			if ( $this->is_ticket_cache_enabled() && ! empty( $cache[ $key ] ) ) {
```

File: src/Tribe/Ticket_Object.php [hunk 4]
```diff
@@ -570,7 +572,7 @@ public function inventory() {
 			$cache = tribe( 'cache' );
 			$key   = __METHOD__ . '-' . $this->ID;
 
-			if ( isset( $cache[ $key ] ) ) {
+			if ( $this->is_ticket_cache_enabled() && isset( $cache[ $key ] ) ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Ticket_Object.php:575]:
```suggestion
			if ( $this->is_ticket_cache_enabled() && ! empty( $cache[ $key ] ) ) {
```

File: src/Tribe/Ticket_Object.php [hunk 5]
```diff
@@ -687,7 +689,7 @@ public function available() {
...
```
Comment (sc0ttkclark) [inline src/Tribe/Ticket_Object.php:692]:
```suggestion
			if ( $this->is_ticket_cache_enabled() && ! empty( $cache[ $key ] ) ) {
```

File: src/Tribe/Ticket_Object.php [hunk 6]
```diff
@@ -729,7 +731,7 @@ public function capacity() {
...
```
Comment (sc0ttkclark) [inline src/Tribe/Ticket_Object.php:734]:
```suggestion
			if ( $this->is_ticket_cache_enabled() && ! empty( $cache[ $key ] ) ) {
```

File: src/Tribe/Ticket_Object.php (hunk 1 above)
Comment (sc0ttkclark) [reply src/Tribe/Ticket_Object.php:1165]:
```suggestion
			 *
			 * @since TBD
			 *
```

File: src/Tribe/Ticket_Object.php (hunk 2 above)
Comment (sc0ttkclark) [reply src/Tribe/Ticket_Object.php:1166]:
```suggestion
			 * @param bool $is_cache_enabled  Whether the ticket has ticket cache enabled.
```

**Review by juanfra — APPROVED** [review]

**Pull request merged**